carapp/authent.py

```diff
@@ -0,0 +1,82 @@
+import flask
+import flask_login
+from . import app, auth, db_models, login_manager, token_serializer
+from flask import make_response, request
+from werkzeug.security import check_password_hash
+
+
+@login_manager.user_loader
+def user_loader(user_id):
+    user = db_models.User.query.get(user_id)
+    if not user:
+        return None
+
+    return user
+
+
+# @login_manager.request_loader
+# def request_loader(request):
+#     username = request.authorization.username
+#     # import pdb; pdb.set_trace()
+#     user = db_models.User.query.filter_by(username=username).first()
+#     if not user:
+#         return None
+
+    # user.is_authenticated = check_password_hash(
+    #    user.password, request.authorization.password
+    # )
+#     return user
+
+
+@auth.verify_token
+def verify_token(token):
+    try:
+        data = token_serializer.loads(token)
+    except:  # noqa: E722
+        return False
+    if 'username' in data:
+        return data['username']
+
+
+@app.route("/")
+def index():
+    if "_user_id" in flask.session:
+        return "Logged in as " + flask_login.current_user.username
+
+    return "You are not logged in!"
+
+
+@app.route("/login", methods=["GET", "POST"])
+def login():
```

Comment:
These guys have to be in the endpoints module since they are endpoints. It is a bit confusing to have a module named endpoints and scattered endpoints over the application modules.

Reply:
Done :)